c/certificateService.c

```diff
@@ -110,159 +111,153 @@ static void respondWithInvalidMethod(HttpResponse *response) {
+            respondWithJsonError(response, "The length of the distinguished name is more than 246 bytes", 400, "Bad Request");
+            goto out;
         }
 
-        userMapStructure.distinguishedNameLength = strlen(distinguishedId);
-        memcpy(userMapStructure.distinguishedName, distinguishedId, strlen(distinguishedId));
-        e2a(userMapStructure.distinguishedName, userMapStructure.distinguishedNameLength);
+        parmlist31->userMapStructure.distinguishedNameLength = strlen(distinguishedId);
+        memcpy(parmlist31->userMapStructure.distinguishedName, distinguishedId, strlen(distinguishedId));
+        e2a(parmlist31->userMapStructure.distinguishedName, parmlist31->userMapStructure.distinguishedNameLength);
         char *registry = jsonObjectGetString(jsonObject, "registry");
-        if (registry == NULL || strlen(registry) == 0) {
-          respondWithBadRequest(response, "registry field not included in request body");
-          return 0;
-        } else if (strlen(registry) > sizeof(userMapStructure.registryName)) {
-          respondWithBadRequest(response, "The length of the registry name is more than 255 bytes");
-          return 0;
+        if(registry == NULL || strlen(registry) == 0) {
+            respondWithJsonError(response, "registry field not included in request body", 400, "Bad Request");
+            goto out;
+        } else if(strlen(registry) > sizeof(parmlist31->userMapStructure.registryName)) {
+            respondWithJsonError(response, "The length of the registry name is more than 255 bytes", 400, "Bad Request");
+            goto out;
         }
 
-        userMapStructure.registryNameLength = strlen(registry);
-        memcpy(userMapStructure.registryName, registry, strlen(registry));
-        e2a(userMapStructure.registryName, userMapStructure.registryNameLength);
+        parmlist31->userMapStructure.registryNameLength = strlen(registry);
+        memcpy(parmlist31->userMapStructure.registryName, registry, strlen(registry));
+        e2a(parmlist31->userMapStructure.registryName, parmlist31->userMapStructure.registryNameLength);
+
+        parmlist31->userMapStructure.functionCode = MAP_DN_TO_USERNAME;
 
-        userMapStructure.functionCode = MAP_DN_TO_USERNAME;
+    } else {
+        respondWithJsonError(response, "Endpoint not found.", 404, "Not Found");
+        goto out;
     }
 
     int rc;
-
 #ifdef _LP64
     __asm(ASM_PREFIX
 	  /* We get the routine pointer for IRRSIM00 by an, *ahem*, direct approach.
 	     These offsets are stable, and this avoids linker/pragma mojo */
-	  " LA 15,X'10' \n"
-          " LG 15,X'220'(,15) \n" /* CSRTABLE */
-          " LG 15,X'28'(,15) \n" /* Some RACF Routin Vector */
-          " LG 15,X'A0'(,15) \n" /* IRRSIM00 itself */
+	        " LLGT 15,X'10' \n"
+          " LLGT 15,X'220'(,15) \n" /* CSRTABLE */
+          " LLGT 15,X'28'(,15) \n" /* Some RACF Routin Vector */
+          " LLGT 15,X'A0'(,15) \n" /* IRRSIM00 itself */
 	  " LG 1,%0 \n"
```

Comment:
Based on the listing, calling of parameter list looks good:
```
* * * * *   S T O R A G E   O F F S E T   L I S T I N G   * * * * *
parmlist31          136-0:131       Class = automatic,           Location = 56(r9),                            Length = 8
```
Code to be executed:
```
  LG       r6,parmlist31(,r9,56)     * From ASM_PREFIX
  LLGT ...                           * 4 x LLGT
  LG       r1,0(r6,)                 * Parameter list
  SAM31
  BALR 14,15
  SAM64
```